From 47cf6ebcb44400ab816c45ea4b9e9ad3e63267a7 Mon Sep 17 00:00:00 2001
From: Tobias <ruck.tobias@gmail.com>
Date: Wed, 20 Nov 2024 18:01:49 +0100
Subject: [PATCH] [DogeCash] Fix `CBlock::GetBlockHeader` for `auxpow` (#55)

Currently, this leaves `auxpow` out, which is incorrect.

(cherry picked from commit e46c93c5353c1134c3d4eaac9fb1d4067f283aad)

diff --git a/src/primitives/block.h b/src/primitives/block.h
index f4543d9aa6e5bb8eee4e2878a356b16f51f5240c..981266b637cc5738495b53f969403b2a9afd3049 100644
--- a/src/primitives/block.h
+++ b/src/primitives/block.h
@@ -96,6 +96,7 @@ public:
         block.nTime = nTime;
         block.nBits = nBits;
         block.nNonce = nNonce;
+        block.auxpow = auxpow;
         return block;
     }
 
diff --git a/src/test/dogecoin_auxpow_methods_tests.cpp b/src/test/dogecoin_auxpow_methods_tests.cpp
index c1c26432df5648b27f63abe5bbbbc97463f696ee..cbaaf0109f7cafcd04694bd2f20783d8e20e2b92 100644
--- a/src/test/dogecoin_auxpow_methods_tests.cpp
+++ b/src/test/dogecoin_auxpow_methods_tests.cpp
@@ -33,12 +33,14 @@ BOOST_AUTO_TEST_CASE(auxpow_block_setnull_test) {
 
     BOOST_CHECK(VersionHasAuxPow(block.nVersion));
     BOOST_CHECK(block.auxpow);
+    BOOST_CHECK(block.GetBlockHeader().auxpow);
 
     // Test SetNull also resets the auxpow
     block.SetNull();
 
     BOOST_CHECK(!VersionHasAuxPow(block.nVersion));
     BOOST_CHECK(!block.auxpow);
+    BOOST_CHECK(!block.GetBlockHeader().auxpow);
 }
 
 BOOST_AUTO_TEST_CASE(auxpow_blockheader_setnull_test) {
